components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -121,6 +122,77 @@ impl CanvasContextState {
     }
 }
 
+#[must_root]
+#[derive(Clone, JSTraceable, MallocSizeOf)]
+pub struct CanvasState {
+    //reflector_: Reflector,
+    #[ignore_malloc_size_of = "Defined in ipc-channel"]
+    ipc_renderer: IpcSender<CanvasMsg>,
+    canvas_id: CanvasId,
+}
+
+impl CanvasState {
+    pub fn new(
+        global: &GlobalScope,
+    ) -> CanvasState {
+        let (sender, receiver) =
+            profiled_ipc::channel(global.time_profiler_chan().clone()).unwrap();
+        let (ipc_renderer, canvas_id) = receiver.recv().unwrap();
```

Comment:
Here we create a new channel, then block waiting on the receiver. This won't work, since nothing sends a value using the sender that can be received.

Reply:
But to create a new paint thread we need a receiver. So how do I deal with it? Is creating a new paint thread necessary?

Reply:
Yes.